build

```
@@ -107,7 +107,7 @@ for PKG_PATH in $(catkin_topological_order --only-folders || colcon list --topol
 
   # Set the version based on the checked out tag that contain at least on digit
   # strip any leading non digits as they are not part of the version number
-  sed -i "1 s@([^)]*)@($( (git describe --tag  --match "*[0-9]*" 2>/dev/null || echo 0) | sed 's@^[^0-9]*@@')-$(date +%Y.%m.%d.%H.%M))@" debian/changelog
+  sed -i "1 s@([^)]*)@($( (git describe --tag  --match "*[0-9]*" 2>/dev/null | sed "s/_/~/" || echo 0) | sed 's@^[^0-9]*@@')-$(date +%Y.%m.%d.%H.%M))@" debian/changelog
```

```suggestion
  sed -i "1 s@([^)]*)@($( (git describe --tag  --match "*[0-9]*" 2>/dev/null || echo 0) | sed -e 's@^[^0-9]*@@' -e '@_@~@g')-$(date +%Y.%m.%d.%H.%M))@" debian/changelog
```

- The sed in the sub shell pipe would result in `echo 0` not being run in case `git describe` finds no matching tags so use the sed that is done anyhow afterwards.
- Also use `g` to replace all `_`.